diff --git a/src/pages/Profile/Profile.jsx b/src/pages/Profile/Profile.jsx
index c013f0b..3e1e99a 100644
--- a/src/pages/Profile/Profile.jsx
+++ b/src/pages/Profile/Profile.jsx
@@ -1,7 +1,10 @@
 import { useState, useEffect } from "react";
 import { useAuth } from "../../context/AuthContext";
 import { getGameSettings, updateGameSettings } from "../../api/auth.api";
-import { generateMouseCommands } from "../../utils/cs2Commands";
+import {
+  generateMouseCommands,
+  generateViewmodelCommands,
+} from "../../utils/cs2Commands";
 
 import toast from "react-hot-toast";
 import SectionTitle from "../../components/Common/SectionsTitle";
@@ -538,6 +541,9 @@ export default function Profile() {
       {/* Section Viewmodel */}
       <div>
         <SectionTitle icon="eye" placeholder="Viewmodel" />
+        <div className="mx-4 mt-6">
+          <CommandCopy commands={generateViewmodelCommands(formData)} />
+        </div>
         <div className="grid grid-cols-2 lg:grid-cols-3 gap-x-10 gap-y-8 mt-10">
           {isEditing ? (
             <>
diff --git a/src/utils/cs2Commands.js b/src/utils/cs2Commands.js
index 2d273db..fc484cf 100644
--- a/src/utils/cs2Commands.js
+++ b/src/utils/cs2Commands.js
@@ -15,6 +15,7 @@ export const generateMouseCommands = (formData) => {
   return commands.length > 0 ? commands.join("; ") : "No settings configured";
 };
 
+// commandes pour le viewmodel
 export const generateViewmodelCommands = (formData) => {
   const commands = [];
 
@@ -22,6 +23,28 @@ export const generateViewmodelCommands = (formData) => {
   if (formData.fov !== null && formData.fov !== "") {
     commands.push(`viewmodel_fov ${formData.fov}`);
   }
+  // offset x
+  if (formData.offsetX !== null && formData.offsetX !== "") {
+    commands.push(`viewmodel_offset_x ${formData.offsetX}`);
+  }
+  // offset y
+  if (formData.offsetY !== null && formData.offsetY !== "") {
+    commands.push(`viewmodel_offset_y ${formData.offsetY}`);
+  }
+  // offset z
+  if (formData.offsetZ !== null && formData.offsetZ !== "") {
+    commands.push(`viewmodel_offset_z ${formData.offsetZ}`);
+  }
+  // hand position
+  if (formData.handPosition !== null && formData.handPosition !== "") {
+    let handValue;
+    if (formData.handPosition === "Right") {
+      handValue = 0;
+    } else if (formData.handPosition === "Left") {
+      handValue = 1;
+    }
+    commands.push(`cl_righthand ${handValue}`);
+  }
 
   return commands.length > 0 ? commands.join("; ") : "No settings configured.";
 };
